docs/development/testing/cmek.md

```diff
@@ -52,18 +61,112 @@
+           
+           # Output Trimmed
+           ...
            diskEncryptionKey:
-               kmsKeyName: projects/projectId/locations/us-east4/keyRings/cmektest/cryptoKeys/alpha/cryptoKeyVersions/2
+               kmsKeyName: projects/projectId/locations/<region>/keyRings/<keyRingName>/cryptoKeys/alpha/cryptoKeyVersions/<revision>
                kmsKeyServiceAccount: user@projectId.iam.gserviceaccount.com
             id: '843731016874110129'
+           ...
        ```
+
+### Gardener Shoot Test
+
+1. Setup a local Gardener shoot cluster. One way to do this is by following instructions at https://github.com/gardener/gardener/blob/master/docs/development/local_setup.md and ensure one installs the GCP extension provider. 
+1. Build and push the images for `garden-extension-provider-gcp` and `machine-controller-manager-provider-gcp` to your docker repo. 
+   1. Inside dir `gardener-extension-provider-gcp`, execute: 
+      - `docker buildx build --push --platform linux/amd64 --tag dockeruser/extension-provider-gcp:latest --target gardener-extension-provider-gcp .`
+   1. In dir: `machine-controller-manager-provider-gcp`, execute:    
+       - `docker buildx build --push --platform linux/amd64 --tag dockeruser/machine-controller-manager-provider-gcp:latest --target machine-controller` 
+1. Change the `ControllerDeployment` yaml for GCP extension provider name: `provider-gcp` by modifying the repository and SHA tag for its image as well as the dependent image of `machine-controller-manager-provider-gcp` to the images pushed above. Apply the controller deployment to the nodeless local garden.
+1. Change the shoot yaml by modifying the `worker` volume and `providerConfig` section as below
+      ```yaml
+         workers:
+            - cri:
+               name: containerd
+            name: worker-1
+            # ... skipped
+            volume:
+               type: pd-balanced
+               size: 50Gi
+            dataVolumes:
+               - name: alpha
+                  type: pd-balanced
+                  size: 25Gi
+               - name: beta
+                  type: SCRATCH
+                  size: 25Gi
+            zones:
+               - europe-west1-c
+            systemComponents:
+               allow: true
+            providerConfig:
+               apiVersion: gcp.provider.extensions.gardener.cloud/v1alpha1
+               kind: WorkerConfig
+               volume:
+                  interface: "SCSI" # applies only for scratch disks. misleading field name
+                  encryption:
+                  kmsKeyName: "projects/projectId/locations/<zoneName>/keyRings/<keyRingName>/cryptoKeys/alpha"
+                  kmsKeyServiceAccount: "user@projectId.iam.gserviceaccount.com" 
+      ```
+1. Apply the shoot YAML agaisnt the `nodeless` cluster and wait till once the shoot has reconciled and the machine(s) in pool are in `Running` phase.
```

Comment:
Apply the shoot YAML against the `nodeless` cluster and wait until the shoot has reconciled and the machine(s) in the worker pool are in `Running` phase.

Reply:
done